sdk/testing/src/main/java/io/opentelemetry/sdk/testing/context/StrictContextStorage.java

```diff
@@ -132,35 +138,28 @@ public Context current() {
    */
   // AssertionError to ensure test runners render the stack trace
   public void ensureAllClosed() {
-    List<CallerStackTrace> leakedCallers = new ArrayList<>();
-    currentCallers.drainTo(leakedCallers);
-    for (CallerStackTrace caller : leakedCallers) {
+    pendingScopes.expungeStaleEntries();
+    for (CallerStackTrace caller : pendingScopes.drainPendingCallers()) {
       // Sometimes unit test runners truncate the cause of the exception.
       // This flattens the exception as the caller of close() isn't important vs the one that leaked
-      AssertionError toThrow =
-          new AssertionError(
-              "Thread [" + caller.threadName + "] opened a scope of " + caller.context + " here:");
-      toThrow.setStackTrace(caller.getStackTrace());
-      throw toThrow;
+      throw callerError(caller);
```

Comment:
so, we're only throwing the first that we run across, rather than alert about all the possibilities?

Reply:
Added logging when multiple, still throwing since this is mostly for unit tests where an exception is best

Reply:
Went ahead and added logging if more than one.